src/pages/Main/Main.jsx

```diff
@@ -1,7 +1,47 @@
 import React from "react";
+import { useSearchParams } from "react-router-dom";
+import List from "./components/List";
+import Map from "./components/Map";
+import styled from "styled-components";
 
 const Main = () => {
-  return <div>메인</div>;
+  const [searchParams, setSearchParams] = useSearchParams();
+  const mapPage = searchParams.get(`page`) === `map`;
+
+  const changePage = () => {
+    if (mapPage) {
+      searchParams.set(`page`, `list`);
+    } else {
+      searchParams.set(`page`, `map`);
+    }
```

Comment:
```suggestion
    const queryValue = mapPage ? "list" : "map";
    searchParams.set(`page`, queryValue);
```
이렇게 구현하면 if 문보다 깔끔해 보입니다~

Reply:
아 이 부분은 스크롤 위치 기억 등 추가 기능 코드가 들어갈 수 있어 일부러 if문으로 했습니다!